src/cascadia/TerminalSettingsEditor/GlobalAppearance.cpp

```diff
@@ -25,5 +25,16 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     void GlobalAppearance::OnNavigatedTo(const NavigationEventArgs& e)
     {
         _ViewModel = e.Parameter().as<Editor::GlobalAppearanceViewModel>();
+        AlwaysShowTabsToggleSwitch().IsEnabled(!_ViewModel.ShowTabsInTitlebar());
+    }
+
+    void GlobalAppearance::ShowTitleBarToggled(const IInspectable& /* sender */, const RoutedEventArgs& /* args */)
+    {
+        AlwaysShowTabsToggleSwitch().IsEnabled(!ShowTitlebarToggleSwitch().IsOn());
+
+        if (ShowTitlebarToggleSwitch().IsOn())
+        {
+            AlwaysShowTabsToggleSwitch().IsOn(true);

```

Comment:
In general, you'll want to edit the setting directly (using `AlwaysShowTabs(true)`) instead of making the switch do it because it reduces code complexity. Right now, this adds a dependency on AlwaysShowTabs being a named switch... but if you use the setting on the ViewModel directly, you can avoid naming the switch at all because the setting is _bound to_ the switch. :smile:

The same thing is true of the `ShowTitlebarToggleSwitch`; if you use the `ShowTitlebar` setting (or whatever it's called,) you can get closer to the data model and avoid permanently naming the switch!